src/opover.c

```diff
@@ -1313,7 +1313,7 @@ Expression *compare_overload(BinExp *e, Scope *sc, Identifier *id)
         if (s_r)
         {
             functionResolve(&m, s_r, e->loc, sc, tiargs, e->e2->type, &args1);
-            if (m.lastf && m.lastf->errors)
+            if (m.lastf && (m.lastf->errors || m.lastf->semantic3Errors))
```

Comment:
From its commit log:

> Instead of preventing CTFE when any error occurred, only give up if an error
> occurred while running semantic3 on the function being called.

@donc had intended to distinguish errors in "function signatures" from the ones in function body.